code/modules/antagonists/wizard/equipment/spellbook.dm

```diff
@@ -553,18 +570,46 @@
 	name = "Curse of Madness"
 	desc = "Curses the station, warping the minds of everyone inside, causing lasting traumas. Warning: this spell can affect you if not cast from a safe distance."
 	cost = 4
+	ritual_invocation = "ALADAL DESINARI ODORI'IN PORES ENHIDO'LEN MORI MAKA TU"
 
 /datum/spellbook_entry/summon/curse_of_madness/Buy(mob/living/carbon/human/user, obj/item/spellbook/book)
 	SSblackbox.record_feedback("tally", "wizard_spell_learned", 1, name)
 	active = TRUE
-	var/message = stripped_input(user, "Whisper a secret truth to drive your victims to madness.", "Whispers of Madness")
-	if(!message)
-		return FALSE
+	var/message
+	while(!message)
+		message = stripped_input(user, "Whisper a secret truth to drive your victims to madness.", "Whispers of Madness")
 	curse_of_madness(user, message)
 	to_chat(user, "<span class='notice'>You have cast the curse of insanity!</span>")
 	playsound(user, 'sound/magic/mandswap.ogg', 50, 1)
 	return TRUE
 
+/datum/spellbook_entry/summon/wild_magic
+	name = "Wild Magic Manipulation"
+	desc = "multiply your remaining spell points by 70%(round down) and expand all of them to Wild Magic Manipulation. \
+		You purchase random spells and items upto the spell points you expanded. Spells from this ritual will no longer be refundable even if you learned it manually, but also the book will no longer accept items to refund."
+	cost = 0
+	ritual_invocation = "ALADAL DESINARI ODORI'IN A'EN SPERMITEN G'ATUA H'UN OVORA DUN SPERMITUN"
+
+/datum/spellbook_entry/summon/wild_magic/Buy(mob/living/carbon/human/user, obj/item/spellbook/book)
+	if(!book.uses)
+		to_chat(user, "<span class='notice'>You have no spell points for this ritual.</span>") // You can cast it again as long as you get more spell points somehow
+		return FALSE
+	SSblackbox.record_feedback("tally", "wizard_spell_learned", 1, name)
+	book.uses = round(book.uses*WIZARD_WILDMAGIC_SPELLPOINT_MULTIPLIER) // more spell points
+	book.refuses_refund = TRUE
+	book.desc = "An unearthly tome that once had a great power."
+	while(book.uses)
+		var/datum/spellbook_entry/target = pick(book.entries)
+		if(istype(target, /datum/spellbook_entry/summon/wild_magic))
+			continue // Too lucky to get more spell points, but no.
+		if(target.CanBuy(user,book))
+			if(target.Buy(user,book))
+				book.uses -= target.cost
```

Comment:
Does this not get subtracted in Buy()?

Reply:
![image](https://user-images.githubusercontent.com/87972842/187893356-0dda424b-1981-4064-a9e0-c3dc7b864733.png)

this image is how you do buy a spell from a spellbook. the proc doesn't increase spell point. and `Buy()` only returns the result if you can 'buy' the spell, so you need to decrease spell points manually by `uses -= cost`